cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TranslatorSuite.scala

```diff
@@ -0,0 +1,405 @@
+
+  test("Break sentence ") {
+    val results = breakSentence.transform(textDf1)
+      .withColumn("sentLen", flatten(col("result.sentLen")))
+      .select("sentLen").collect()
+    val headStr = results.head.getSeq(0).mkString("\n")
+    assert(headStr === "25")
+  }
+
+  override def testObjects(): Seq[TestObject[BreakSentence]] =
+    Seq(new TestObject(breakSentence, textDf1))
+
+  override def reader: MLReadable[_] = BreakSentence
+}
+
+class DictionaryLookupSuite extends TransformerFuzzing[DictionaryLookup]
+  with TranslatorKey with Flaky with TranslatorUtils {
+
+  import spark.implicits._
+
+  lazy val dictDf: DataFrame = Seq(List("fly")).toDF("text")
+
+  lazy val dictionaryLookup: DictionaryLookup = new DictionaryLookup()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setFromLanguage("en")
+    .setToLanguage("es")
+    .setTextCol("text")
+    .setOutputCol("result")
+
+  test("Break sentence ") {
+    val results = dictionaryLookup.transform(dictDf)
+      .withColumn("translations", flatten(col("result.translations")))
+      .withColumn("normalizedTarget", col("translations.normalizedTarget"))
+      .select("normalizedTarget").collect()
+    val headStr = results.head.getSeq(0).mkString("\n")
+    assert(headStr === "volar\nmosca\noperan\npilotar\nmoscas\nmarcha")
+  }
+
+  override def testObjects(): Seq[TestObject[DictionaryLookup]] =
+    Seq(new TestObject(dictionaryLookup, dictDf))
+
+  override def reader: MLReadable[_] = DictionaryLookup
+}
+
+class DictionaryExamplesSuite extends TransformerFuzzing[DictionaryExamples]
+  with TranslatorKey with Flaky with TranslatorUtils {
+
+  import spark.implicits._
+
+  lazy val dictDf: DataFrame = Seq(List(("fly", "volar"))).toDF("textAndTranslation")
+
+  lazy val dictionaryExamples: DictionaryExamples = new DictionaryExamples()
+    .setSubscriptionKey(translatorKey)
+    .setLocation("eastus")
+    .setFromLanguage("en")
+    .setToLanguage("es")
+    .setTextAndTranslationCol("textAndTranslation")
+    .setOutputCol("result")
+
+  test("Dictionary Examples") {
+    val results = dictionaryExamples.transform(dictDf)
+      .withColumn("examples", flatten(col("result.examples")))
+      .withColumn("sourceTerm", col("examples.sourceTerm"))
+      .withColumn("targetTerm", col("examples.targetTerm"))
+      .select("sourceTerm", "targetTerm").collect()
+    assert(results.head.getSeq(0).head.toString === "fly")
+    assert(results.head.getSeq(1).head.toString === "volar")
+  }
+
+  override def testObjects(): Seq[TestObject[DictionaryExamples]] =
+    Seq(new TestObject(dictionaryExamples, dictDf))
+
+  override def reader: MLReadable[_] = DictionaryExamples
+}
+
+class DocumentTranslatorSuite extends TransformerFuzzing[DocumentTranslator]
+  with TranslatorKey with Flaky {
+
+  import spark.implicits._
+
+  // TODO: Replace root SAS urls after 2022-07-07
+  lazy val sourceRoot: String = "?sp=rl&st=2021-07-06T06:28:26Z&se=2022-07-07T06:28:00Z" +
```

Comment:
might want to consider factor this in the following way

lazy val containerSasToken = ... (If this needs to be a SAS otherwise we should try to remove SAS)
lazy val urlRoot = "https://mmlspark.blob.core.windows.net/"

and use the same container as for all experiments. If you find the container SAS to be used all over we might consider breaking this into a single trait for all cog service tests so that we only need to update in one location if it expires